okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.java

```diff
@@ -99,11 +90,31 @@ public CallServerInterceptor(boolean forWebSocket) {
         .responseHeadersEnd(realChain.call(), response);
 
     int code = response.code();
+    if (code == 100) {
+      // server sent a 100-continue even though we did not request one. Send the request body again.
```

Comment:
This doesn't seem to match the RFC which states

> The client ought to continue sending the request and discard the 100 response.

Specifically, why are we resending the request body? It seems like we should ignore the 100 and just try to read a later HTTP response?

Reply:
Good point. I will try if it works without resending the body.